### PR TITLE
first steps to upload to internal

### PR DESCRIPTION
**Pull request by leandrobono**
## Ticket

PLAT-7821
_Related tickets:_
_Related PRs:_

## Type of PR

- [ ] Bugfix
- [X] New feature
- [ ] Minor changes

## Changes introduced

This feature lets the user deploy automatically an android app to the internal test track on the google play store

### CODE REVIEW

**Review by danielsousaio — CHANGES_REQUESTED** [review]
All scaffold customizations should be made at [`scaffold/template/custom`](https://github.com/crowdbotics/modules/tree/master/scaffold/template/custom). You can add new files too, if needed. Just make sure to follow the same directories structure that the template uses.

When adding new files, followup with an update to [scaffold/install.js](https://github.com/crowdbotics/modules/blob/master/scaffold/install.js) too. This is how we programmatically override files in a React Native template. This strategy is helpful for doing versions upgrade.

**Comment by leandrobono** [comment]
@danielsousaio i made changes on the custom folder for fastlane, but i also needed to make changes on the android native files. How can i do them on the custom folder? Cause they live on scaffold/template/android

File: scaffold/template/custom/android/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+            storePassword 'android'
+            keyAlias 'androiddebugkey'
+            keyPassword 'android'
+        }
+        release {
+            storeFile file('my-upload-key.keystore')
+            storePassword 'my-key'
+            keyAlias 'my-key-alias'
+            keyPassword 'my-key'
+        }
```
Comment (danielsousaio) [inline scaffold/template/custom/android/app/build.gradle:160]: keystores are private, we can't include them in the scaffold

https://reactnative.dev/docs/signed-apk-android

File: scaffold/template/custom/android/build.gradle [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+
+buildscript {
+    ext {
+        buildToolsVersion = "29.0.3"
+        minSdkVersion = 21
+        compileSdkVersion = 30
+        targetSdkVersion = 30
```
Comment (danielsousaio) [inline scaffold/template/custom/android/build.gradle:8]: This bumps the target to Android 11 right? Why do we need to change this? 

I'm wondering if there's any potential issues with this change.

Comment (leandrobono) [reply scaffold/template/custom/android/build.gradle:8]: i dont remember the exact fastlane error right now, but it was asking to upgrade the sdk version. Do you mean we can have errors with previous built apps?

**Comment by mabdullahabid** [comment]
@danielsousaio why does this branch look like it has a lot of stuff from other PRs and branches? Should we just get @leandrobono to create a new one with only his changes?

**Review by sgmagar — COMMENTED** [review]
Requires some update to deal with the flow mainly the manual steps required.

**Comment by danielsousaio** [comment]
Closing stale PR

**Pull request closed without merging**